src/prefect/storage/azure.py

```diff
@@ -40,20 +43,18 @@ class Azure(Storage):
     def __init__(
         self,
         container: str,
-        connection_string: str = None,
+        connection_string_secret_name: str = None,
         blob_name: str = None,
         stored_as_script: bool = False,
         **kwargs: Any
     ) -> None:
-        self.connection_string = connection_string or os.getenv(
-            "AZURE_STORAGE_CONNECTION_STRING"
-        )
-
         self.container = container
+        self.connection_string_secret_name = connection_string_secret_name
         self.blob_name = blob_name
 
         result = AzureResult(
-            connection_string=self.connection_string, container=container
+            connection_string_secret=self.connection_string_secret_name,
```

Comment:
`AzureResult` takes in `connection_string_secret`. Do we want the `AzureStorage` class to take in `connection_string_secret` instead of `connection_string_secret` name so that we have a uniform interface?